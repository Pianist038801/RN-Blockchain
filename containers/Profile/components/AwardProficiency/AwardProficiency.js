import React from 'react'
import PropTypes from 'prop-types'
import { View, Text, Dimensions } from 'react-native'
import { Progress } from 'nachos-ui'
import _find from 'lodash/find'
import proficiencyOptions from '../../utils/languagesProficiency.js'
import styles from '../../../../style.js'

const AwardProficiency = (props) => {
  const { award, range, type } = props
  let proficiency = parseInt(award.proficiency, 10)
  proficiency = (proficiency > range) ? range : proficiency
  let secondaryLabel = proficiency
  if (type === 'language') {
    secondaryLabel = _find(proficiencyOptions, {value: proficiency}).label
  }
  const { width } = Dimensions.get('window')

  return (
    <View style={styles.profileEntityContainer}>
      <View style={styles.profileEntityContent}>
        <Text style={styles.profileEntityPrimaryText}>
          {award.name}
        </Text>
        <Text style={styles.profileEntitySecondaryText}>
          {secondaryLabel}
        </Text>
        <Progress
          progress={(1 / range) * parseInt(proficiency, 10)}
          width={width * 0.65}
        />
      </View>
    </View>
  )
}

AwardProficiency.defaultProps = {
  range: 5
}

AwardProficiency.propTypes = {
  award: PropTypes.shape({
    name: PropTypes.string,
    proficiency: PropTypes.oneOfType([
      PropTypes.string.isRequired,
      PropTypes.number.isRequired
    ]),
  }).isRequired,
  range: PropTypes.number,
  type: PropTypes.string.isRequired
}

export default AwardProficiency
