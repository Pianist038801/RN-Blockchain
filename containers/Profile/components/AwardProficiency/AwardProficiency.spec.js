import React from 'react'
import renderer from 'react-test-renderer'
import AwardProficiency from './index'

describe('containers/Profile/components/AwardProficiency', () => {
  describe('when range use defaultProps value', () => {
    it('renders correctly', () => {
      const props = {
        award: {
          name: 'Skill1',
          proficiency: 3
        },
        type: 'skill'
      }

      const tree = renderer.create(
        <AwardProficiency {...props} />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })

  describe('when props.range = 11', () => {
    it('renders correctly', () => {
      const props = {
        award: {
          name: 'English',
          proficiency: 6,
        },
        type: 'language',
        range: 11
      }

      const tree = renderer.create(
        <AwardProficiency {...props} />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })
})
