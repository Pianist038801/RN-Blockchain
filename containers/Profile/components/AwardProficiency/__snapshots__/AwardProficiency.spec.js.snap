// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/Profile/components/AwardProficiency when props.range = 11 renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#fff",
      "borderWidth": 1,
      "flex": 0.5,
      "flexDirection": "row",
      "marginBottom": 5,
      "padding": 22.5,
      "width": 600,
    }
  }
>
  <View
    style={
      Object {
        "marginLeft": 22.5,
        "width": 360,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#569cee",
          "fontWeight": "bold",
        }
      }
    >
      English
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#000",
          "marginBottom": 5,
          "marginTop": 5,
          "width": 750,
        }
      }
    >
      Intermediate - High (B1)
    </Text>
    <View
      style={
        Object {
          "backgroundColor": "#bdc1cc",
          "borderRadius": 3,
          "height": 6,
          "overflow": "hidden",
          "width": 487.5,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#2f8cff",
            "borderRadius": 3,
            "height": 6,
            "width": 265.9090909090909,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`containers/Profile/components/AwardProficiency when range use defaultProps value renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#fff",
      "borderWidth": 1,
      "flex": 0.5,
      "flexDirection": "row",
      "marginBottom": 5,
      "padding": 22.5,
      "width": 600,
    }
  }
>
  <View
    style={
      Object {
        "marginLeft": 22.5,
        "width": 360,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#569cee",
          "fontWeight": "bold",
        }
      }
    >
      Skill1
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#000",
          "marginBottom": 5,
          "marginTop": 5,
          "width": 750,
        }
      }
    >
      3
    </Text>
    <View
      style={
        Object {
          "backgroundColor": "#bdc1cc",
          "borderRadius": 3,
          "height": 6,
          "overflow": "hidden",
          "width": 487.5,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#2f8cff",
            "borderRadius": 3,
            "height": 6,
            "width": 292.50000000000006,
          }
        }
      />
    </View>
  </View>
</View>
`;
